src/Protocol/Protocol18x.cpp

```diff
@@ -1075,7 +1075,7 @@ void cProtocol180::SendRespawn(eDimension a_Dimension, bool a_ShouldIgnoreDimens
 	if ((m_LastSentDimension == a_Dimension) && !a_ShouldIgnoreDimensionChecks)
 	{
 		// Must not send a respawn for the world with the same dimension, the client goes cuckoo if we do (unless we are respawning from death)
-		return;
```

Comment:
Based on @Pokechu22's comment in #3114, https://github.com/cuberite/cuberite/pull/3114#discussion_r60810597, I don't think we can remove this line. However I do think we should follow his advice, and send two respawns in this situation, one to a different dimension, and one back. We should also rename `a_ShouldIgnoreDimensionChecks` to `a_SpawningInSameWorld` and make the change in cProtocol17x.cpp as well.